front-end/orderEditor.py

```diff
@@ -0,0 +1,148 @@
+
+        buttons_frame = tk.Frame(self)
+        buttons_frame.grid(row=2, column=0, columnspan=2, pady=10)
+
+        tk.Button(buttons_frame, text="Zapisz Zamówienie", font=("Arial", 14), bg="green", fg="white",
+                  command=self.save_order).pack(side="left", padx=10)
+
+        (tk.Button(buttons_frame, text="Anuluj", font=("Arial", 14), bg="red", fg="white",
+                   command=lambda: self.manager.switch_to("TabletopDashboard",
+                                                          table_name=self.table_name,
+                                                          table_id=self.manager.current_table_id)).pack(side="right",
+                                                                                                        padx=10))
+
+    def update_meal_list(self, *args):
+        search_term = self.search_var.get().lower()
+        search_term = self.search_var.get().lower()
+        self.meal_listbox.delete(0, tk.END)
+        for meal in self.meals:
+            if search_term in meal['name'].lower():
+                self.meal_listbox.insert(tk.END, f"{meal['meal_id']}: {meal['name']} - {meal['price']} PLN")
+
+    def add_selected_meal(self, event=None):
+        selection = self.meal_listbox.curselection()
+        if selection:
+            index = selection[0]
+            meal_text = self.meal_listbox.get(index)
+            meal_id = int(meal_text.split(":")[0])
+            meal = next((m for m in self.meals if m['meal_id'] == meal_id), None)
+            if meal:
+                self.selected_meals.append(meal)
+                self.update_order_meals_listbox()
+
+    def remove_selected_meal(self, event=None):
+        selection = self.order_meals_listbox.curselection()
+        if selection:
+            index = selection[0]
+            del self.selected_meals[index]
+            self.update_order_meals_listbox()
+
+    def update_order_meals_listbox(self):
+        self.order_meals_listbox.delete(0, tk.END)
+        for meal in self.selected_meals:
+            self.order_meals_listbox.insert(tk.END, f"{meal['name']} - {meal['price']} PLN")
+
+    def load_order_meals(self):
+        self.selected_meals = self.order.get('meals', [])
+        self.update_order_meals_listbox()
+
+    def save_order(self):
+        if not self.selected_meals:
+            messagebox.showwarning("Uwaga", "Zamówienie nie może być puste.")
+            return
+
+        ORDERS_FILE = os.path.join(os.path.dirname(os.path.abspath(__file__)), "orders.json")
+        if os.path.exists(ORDERS_FILE):
+            with open(ORDERS_FILE, "r", encoding="utf-8") as file:
+                orders = json.load(file)
+        else:
+            orders = []
+
+        if self.order:
+            for o in orders:
+                if o['order_id'] == self.order['order_id']:
+                    o['meals'] = self.selected_meals
+                    break
+            messagebox.showinfo("Sukces", "Zamówienie zaktualizowane.")
```

Comment:
Kolejny msgbox

Reply:
ta message boxy zostały przy zamowieniach bo to wtedy informacje musial bym dac by wysiwetlila sie w tabetopdashboard - ale wygląda przejzyscie